OverlayPlugin.Core/MemoryProcessors/ContentFinderSettings/LineContentFinderSettings.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Advanced_Combat_Tracker;
+
+namespace RainbowMage.OverlayPlugin.MemoryProcessors.ContentFinderSettings
+{
+    class LineContentFinderSettings
+    {
+        public const uint LogFileLineID = 265;
+        private ILogger logger;
+        private readonly FFXIVRepository ffxiv;
+
+        private Func<string, DateTime, bool> logWriter;
+
+        private IContentFinderSettingsMemory contentFinderSettingsMemory;
+
+        private bool wroteFirstLine = false;
+
+        public LineContentFinderSettings(TinyIoCContainer container)
+        {
+            logger = container.Resolve<ILogger>();
+            ffxiv = container.Resolve<FFXIVRepository>();
+            if (!ffxiv.IsFFXIVPluginPresent())
+                return;
+            contentFinderSettingsMemory = container.Resolve<IContentFinderSettingsMemory>();
+            var customLogLines = container.Resolve<FFXIVCustomLogLines>();
+            this.logWriter = customLogLines.RegisterCustomLogLine(new LogLineRegistryEntry()
+            {
+                Name = "ContentFinderSettings",
+                Source = "OverlayPlugin",
+                ID = LogFileLineID,
+                Version = 1,
+            });
+            ActGlobals.oFormActMain.BeforeLogLineRead += LogLineHandler;
+        }
+
+        private void LogLineHandler(bool isImport, LogLineEventArgs args)
```

Comment:
This makes sense, I can adjust to this.

Reply:
Hmm, this would cause issues with forcing a log line on ACT startup if OP's startup process is slow enough that the event fires before this handler is registered, wouldn't it?

I work around this by just checking every line right now.

Reply:
The "force log line" logic wasn't really compatible with adding zone ID/name to the line, so I just ditched it, which should be fine anyways because if it breaks here due to slow load/registering it'll break a few other places as well.

Reply:
I'm not aware of a way to get the zone ID, only the zone name (from ACT itself). As far as I can see there doesn't seem to be a method exposed on the FFXIV_ACT_Plugin interfaces to get this information.

(There's a `GetCurrentTerritoryID` but that's not the same)

Reply:
Isn't territory id the same? The content finder condition might have a more specific name, but the airquotes "zone id" is the territory id, unless I'm horribly misremembering how this all works.